### ISSUE TITLE
Summit Analysis Robustness.

### ISSUE THREAD

**Issue by Marchowes** [Marchowes/pyProm#1]
Right now, the only criteria for determining if a gridpoint is a summit or not is if no neighbor is higher.

This is insufficient since neighbors of equal height can have non-common neighbors of greater height, which disqualifies the cluster of equal height neighbors. Routines for analysis need to be written.

**Comment by Marchowes** [comment]
-Write Tree building class.  Trees are built from equal height neighbors.
-Tree members are exempt from further analysis as analysis is costly.
-Tree member checking is also costly, so exemption checking occurs when a grid point is found to have an equal height neighbor.
-Trees will have types. Summit, NonSummit, Col, maybe "pond"
-Bonus: for Summit types, and Col types, determine the middle of the set. This may be superfluous.
---
Ponds/waterbodies are the worst violators of the equal height neighbor issue.

**Comment by Marchowes** [comment]
So, turns out Tree building easily breaks Python's Maximum recursion, so instead I've made a less fancy EqualHeightBlob routine which can be used to dump a large List of coordinates into a Blob Object.

I have a functioning proof-of-concept, but the point exemption lookup was very slow on last night's run. It barely got 20% through a map after about 8 hours. Lookup is probably slow becasue its checking for exemptions in a very large list. I suspect creating a {y:x} hash will speed lookups up quite a bit. 

Blob creation is also not the fastest, I did a test on Moore Reservoir, which analysis found about 12,000 points and it took about 30 seconds.
